### PR TITLE
Implement enabling/disabling schedulers

### PR DESCRIPTION
**Pull request by jpbruinsslot**
### Changes

Added the option to disable and enable schedulers. With a PATCH requests on the scheduler endpoint a scheduler can be enabled or disabled by setting the value of `enabled` in the request. The scheduler will stop all threads that populate the queue with items, and will set all the tasks that have the status `QUEUED` in the the database to `CANCELLED`.

We don't remove the scheduler (from memory) since this is reserved for when organisation are removed. We want to be able to re-enable the scheduler when this is requested.

### Issue link

https://github.com/minvws/nl-kat-coordination/issues/1166

### CODE REVIEW

**Review by Donnype — APPROVED** [review]

File: mula/tests/integration/test_boefje_scheduler.py [hunk 1]
```diff
@@ -624,6 +625,57 @@ def test_post_pop(self):
         self.assertEqual(task_db.id, p_item.id)
         self.assertEqual(task_db.status, models.TaskStatus.DISPATCHED)
 
+    def test_disable_scheduler(self):
+        # Act
+        self.scheduler.disable()
```
[the remark this answers is not in the document]
Comment (jpbruinsslot) [reply mula/tests/integration/test_boefje_scheduler.py:630]: Yeah I initially created it for a BoefjeScheduler, however I wanted to make sure that the procedures could work for any particular scheduler. Then I was left with the decision to remove this, and thought it wouldn't be bad in order to leave it in for indeed more coverage. But it does introduce some duplication.

File: mula/scheduler/repositories/stores.py [hunk 2]
```diff
@@ -18,8 +18,12 @@ def __init__(self) -> None:
     @abc.abstractmethod
     def get_tasks(
         self,
-        scheduler_id: str,
-        filters: Optional[List[models.Filter]],
+        scheduler_id: Optional[str] = None,
```
[the remark this answers is not in the document]
Comment (jpbruinsslot) [reply mula/scheduler/repositories/stores.py:21]: Afaik, it won't break previous usages. This PR introduced starting using this method in the scheduler

File: mula/scheduler/models/scheduler.py [hunk 3]
```diff
@@ -8,5 +8,5 @@ class Scheduler(BaseModel):
     unmarshalling of schedulers to a JSON representation."""
 
     id: Optional[str]
-    populate_queue_enabled: Optional[bool]
+    enabled: Optional[bool]
```
Comment (ammar92) [inline mula/scheduler/models/scheduler.py:11]: I was wondering why this property is optional. What's the expected behavior if it's unset?

**Review by ammar92 — APPROVED** [review]
Looks good to me. Just a single remark/ question.

File: mula/scheduler/models/scheduler.py (hunk 3 above)
Comment (jpbruinsslot) [reply mula/scheduler/models/scheduler.py:11]: This optional is mainly used for the serialization of the model when patching. Might need to think about what models are used as a database representation and what are used for REST response models, and have a clearer definition and separation of those.

Comment (ammar92) [reply mula/scheduler/models/scheduler.py:11]: If you're using this entire structure/ entity in the `PATCH` request, then you might want to consider having a default value instead.

Comment (jpbruinsslot) [reply mula/scheduler/models/scheduler.py:11]: No not the entire structure. Think the only viable field for patching is `enabled` at the moment

**Comment by Darwinkel** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
* Sending enable True/False PATCH request results in the container reporting a disabled scheduler:
image


(Rocky now gives the following error when trying to run a Boefje from the katalogus:)
image

File: mula/scheduler/models/scheduler.py (hunk 3 above)
Comment (ammar92) [reply mula/scheduler/models/scheduler.py:11]: Yes, but the current endpoint also allows `{"enabled": null}" which doesn't do anything which is unusual for a `PATCH` request.

Different representations are indeed a good idea. It's also common to have different request and response models. I'll leave it up to you for now.

Comment (jpbruinsslot) [reply mula/scheduler/models/scheduler.py:11]: Ah I see. I'll have to look into it if this relates to the omission of the field (key/value) in a patch request or the omission of the value. But I'll up the suggestion of the different representations in another issue, which can be a nice addition to make the differentiation more clear.

Comment (jpbruinsslot) [reply mula/scheduler/models/scheduler.py:11]: > Yes, but the current endpoint also allows
> 
> ```json
> {"enabled": null}
> ```
> 
> which doesn't do anything which is unusual for a `PATCH` request.
> 
> Different representations are indeed a good idea. It's also common to have different request and response models. I'll leave it up to you for now.

that json request will not be allowed by the current endpoint.

```
{
    "detail": [
        {
            "loc": [
                "body",
                "enabled"
            ],
            "msg": "value could not be parsed to a boolean",
            "type": "type_error.bool"
        }
    ]
}
```

**Review by underdarknl — APPROVED** [review]

**Pull request merged**